src/main/java/com/wezaam/withdrawal/application/impl/WithdrawalServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+package com.wezaam.withdrawal.application.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.wezaam.withdrawal.application.WithdrawalProcessorService;
+import com.wezaam.withdrawal.application.WithdrawalService;
+import com.wezaam.withdrawal.domain.command.CreateWithdrawalCommand;
+import com.wezaam.withdrawal.domain.entity.User;
+import com.wezaam.withdrawal.domain.entity.Withdrawal;
+import com.wezaam.withdrawal.domain.entity.WithdrawalScheduled;
+import com.wezaam.withdrawal.domain.event.WithdrawalEvent;
+import com.wezaam.withdrawal.domain.exception.NotFoundException;
+import com.wezaam.withdrawal.domain.ports.output.repository.UserRepository;
+import com.wezaam.withdrawal.domain.ports.output.repository.WithdrawalRepository;
+import com.wezaam.withdrawal.domain.ports.output.repository.WithdrawalScheduledRepository;
+import com.wezaam.withdrawal.domain.valueobject.WithdrawalType;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class WithdrawalServiceImpl implements WithdrawalService {
+
+    private final WithdrawalRepository withdrawalRepository;
+
+    private final WithdrawalScheduledRepository withdrawalScheduledRepository;
+
+    private final UserRepository userRepository;
+
+    private final WithdrawalProcessorService withdrawalProcessorService;
+
+    @Override
+    @Transactional
+    public Withdrawal create(CreateWithdrawalCommand createWithdrawalCommand) {
+
+        User user =
+                userRepository
+                        .findById(createWithdrawalCommand.getUserId())
+                        .orElseThrow(() -> new NotFoundException("User not found"));
+
+        if (WithdrawalType.ASAP.name().equals(createWithdrawalCommand.getType().name())) {
```

Comment:
Hey @gasperlf, have you consider megring `Withdrawal` and `WithdrawalScheduled`?

Reply:
Yes, I did.
but at the end I've prioritize code refactor  to keep functionality and then go with other changes.

Reply:
And with the new architecture base on domain it is possible to extend o segregate withdrawal

Reply:
Ok, let's discuss it tomorrow :)